### ISSUE TITLE
class  ignored keyword [this] 

### ISSUE THREAD

**Issue by some-code** [Microsoft/TypeScript#25113]
**TypeScript Version:**  3.0.0-dev.20180620 and 2.9.2
**Code**

```ts
class Animal {
  name: string
  constructor(name: string) {
    this.name = name;
  }
  sayHi(): string {
    return `my name is ${name}`
  }
}
```

**Expected behavior:**
`sayHi` method should throw an error, the `name`  is not found

**Actual behavior:**
it is all ok

**Playground Link:** 
[showcase](https://www.typescriptlang.org/play/index.html#src=class%20Animal%20%7B%0D%0A%20%20name%3A%20string%0D%0A%20%20constructor(name%3A%20string)%20%7B%0D%0A%20%20%20%20this.name%20%3D%20name%3B%0D%0A%20%20%7D%0D%0A%20%20sayHi()%3A%20string%20%7B%0D%0A%20%20%20%20return%20%60my%20name%20is%20%24%7Bname%7D%60%0D%0A%20%20%7D%0D%0A%7D) 

**Related Issues:**  
none

**Comment by some-code** [comment]
thank you!  
I understand
